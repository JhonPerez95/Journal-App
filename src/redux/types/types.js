export const types = {
  login: '[Auth] login',
  logout: '[Auth] logout',

  uiSetError: '[UI] Set Error',
  uiRemoveError: '[UI] Remove Error',
  uiStartLoading: '[UI] Start loading',
  uiFinishLoading: '[UI] Finish loading',

  notesAddNew: '[Notes] New Note',
  notesActive: '[Notes] Active Note',
  notesLoad: '[Notes] Load Notes',
  notesUpdate: '[Notes] Updated Notes',
  notesFileUrl: '[Notes] Updated Img Url',
  notesDelete: '[Notes] Updated Img Url',
  notesLogoutCleaning: '[Notes] Logout Cleanning',
};
