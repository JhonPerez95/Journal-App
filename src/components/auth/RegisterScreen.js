import React from 'react';
import { Link } from 'react-router-dom';
import validator from 'validator';

import { useDispatch } from 'react-redux';

import useForm from '../../hook/useForm';
import { actRemoveError, actAlertError } from '../../redux/actions/uiAction';
import { startEmailRegister } from '../../redux/actions/authActions';

const RegisterScreen = () => {
  const dispatch = useDispatch();
  // const { msgError } = useSelector((state) => state.ui);

  const [formValue, handleInputChange] = useForm({
    name: '',
    email: '',
    password: '',
    password2: '',
  });
  const { name, email, password, password2 } = formValue;

  const handleSubmit = (e) => {
    e.preventDefault();
    if (isFormValid()) {
      dispatch(startEmailRegister(email, password, name));
    }
  };

  const isFormValid = () => {
    if (name.trim().length === 0) {
      dispatch(actAlertError('Name is required'));
      return false;
    } else if (!validator.isEmail(email)) {
      dispatch(actAlertError('Email is not valid'));
      return false;
    } else if (password !== password2 || password.length <= 5) {
      dispatch(actAlertError('Password is not valid'));
      return false;
    }
    dispatch(actRemoveError());
    return true;
  };

  return (
    <>
      <h3 className="auth__title">Registrer</h3>
      {/* {msgError && <div className="auth__alert-error">{msgError}</div>} */}
      <form onSubmit={handleSubmit}>
        <input
          className="auth__input"
          type="text"
          name="name"
          placeholder="Name"
          value={name}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="text"
          name="email"
          placeholder="Email"
          value={email}
          onChange={handleInputChange}
          autoComplete="off"
        />
        <input
          className="auth__input"
          type="password"
          name="password"
          placeholder="Password"
          value={password}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          name="password2"
          placeholder="Confirm"
          value={password2}
          onChange={handleInputChange}
        />
        <button className="btn btn-primary btn-block mb-5" type="submit">
          Register
        </button>

        <Link className="link" to="/auth/login">
          Already registered?
        </Link>
      </form>
    </>
  );
};

export default RegisterScreen;
