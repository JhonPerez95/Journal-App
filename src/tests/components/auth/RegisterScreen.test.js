import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store'; //ES6 modules
import { MemoryRouter } from 'react-router-dom';

import RegisterScreen from '../../../components/auth/RegisterScreen';
import { types } from '../../../redux/types/types';

// Config Store Redux
const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initState = { auht: {}, ui: { loading: false, msgError: null } };
let store = mockStore(initState);
// store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <RegisterScreen />
    </MemoryRouter>
  </Provider>
);

describe('Test the component ', () => {
  test('should to render <RegisterScreen /> ', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should call actions ', () => {
    const emailField = wrapper.find('input[name="email"]');

    emailField.simulate('change', {
      target: {
        value: '',
        name: 'email',
      },
    });
    wrapper.find('form').prop('onSubmit')({
      preventDefault() {},
    });

    const resAction = store.getActions();
    // console.log(resAction);
    expect(resAction[0]).toEqual({
      type: types.uiSetError,
      payload: 'Name is required',
    });
  });
});
