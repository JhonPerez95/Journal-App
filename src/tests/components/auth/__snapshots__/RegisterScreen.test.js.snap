// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the component  should to render <RegisterScreen />  1`] = `
Array [
  <h3
    className="auth__title"
  >
    Registrer
  </h3>,
  <form
    onSubmit={[Function]}
  >
    <input
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value=""
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block mb-5"
      type="submit"
    >
      Register
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already registered?
    </a>
  </form>,
]
`;
